libraries/core-react/src/components/Menu/Menu.tsx

```diff
@@ -28,15 +28,6 @@ const MenuPaper = styled(Paper)<MenuPaperProps>`
   min-width: fit-content;
   ${bordersTemplate(border)};
   ${({ open }) => css({ visibility: open ? null : 'hidden' })}
-
-  li:first-child {
-    border-top-left-radius: ${border.type === 'border' && border.radius};
-    border-top-right-radius: ${border.type === 'border' && border.radius};
-  }
-  li:last-child {
```

Comment:
This border-radius hack is no longer needed after spacing was re-introduced to Menu